src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -114,6 +131,7 @@ func (h *PromReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 // ServeHTTPWithEngine returns query results from the storage
 func (h *PromReadHandler) ServeHTTPWithEngine(w http.ResponseWriter, r *http.Request, engine *executor.Engine) ([]*ts.Series, models.RequestParams, *RespError) {
 	ctx := context.WithValue(r.Context(), handler.HeaderKey, r.Header)
+	sp := opentracingutil.SpanFromContextOrRoot(ctx)
```

Comment:
Might be better to just put this in the error handler, since it's not used aside from error reporting so there's little reason to create it on every request